distro.py

```diff
@@ -983,11 +983,20 @@ def _get_distro_release_info(self):
                 distro_info['id'] = match.group(1)
             return distro_info
         else:
-            basenames = os.listdir(_UNIXCONFDIR)
-            # We sort for repeatability in cases where there are multiple
-            # distro specific files; e.g. CentOS, Oracle, Enterprise all
-            # containing `redhat-release` on top of their own.
-            basenames.sort()
+            try:
+                basenames = os.listdir(_UNIXCONFDIR)
+                # We sort for repeatability in cases where there are multiple
+                # distro specific files; e.g. CentOS, Oracle, Enterprise all
+                # containing `redhat-release` on top of their own.
+                basenames.sort()
+            except OSError:
+                # This may occur when /etc is not readable but we can't be
+                # sure about the *-release files. Check common entries of
+                # /etc for information. If they turn out to not be there the
+                # error is handled in `_parse_distro_release_file()`.
+                basenames = ['SuSE-release', 'base-release', 'centos-release',
```

Comment:
Actually, you can ignore anything in `_DISTRO_RELEASE_IGNORE_BASENAMES`, so you can remove oem-release, system-release, lsb-release, debian_version and os-release :) So I guess we're left with

```python
basenames = [''SuSE-release', 'base-release', 'centos-release', 'fedora-release', 'redhat-release', 'arch-release', 'gentoo-release', 'manjaro-release', 'oracle-release', 'sl-release', slackware-version']
```